sentry-android-core/src/main/java/io/sentry/android/core/internal/util/AndroidConnectionStatusProvider.java

```diff
@@ -31,104 +34,402 @@
 public final class AndroidConnectionStatusProvider implements IConnectionStatusProvider {
 
   private final @NotNull Context context;
-  private final @NotNull ILogger logger;
+  private final @NotNull SentryOptions options;
   private final @NotNull BuildInfoProvider buildInfoProvider;
+  private final @NotNull ICurrentDateProvider timeProvider;
   private final @NotNull List<IConnectionStatusObserver> connectionStatusObservers;
   private final @NotNull AutoClosableReentrantLock lock = new AutoClosableReentrantLock();
   private volatile @Nullable NetworkCallback networkCallback;
 
+  private static final @NotNull AutoClosableReentrantLock connectivityManagerLock =
+      new AutoClosableReentrantLock();
+  private static volatile @Nullable ConnectivityManager connectivityManager;
+
+  private static final int[] transports = {
```

Comment:
I'd add `TRANSPORT_BLUETOOTH` here, e.g. for wearables without cellular connection.